### ISSUE TITLE
[FP]: False positive against bcprov-jdk15on for CVE_2020-0187

### ISSUE THREAD

**Issue by jubui** [dependency-check/DependencyCheck#5539]
### Package URl

`pkg:maven/org.bouncycastle/bcprov-jdk15on@1.64`

### CPE

`cpe:2.3:a:org.bouncycastle:bcprov-jdk15on:1.64:*:*:*:*:*:*:*`

### CVE

CVE-2020-0187

### ODC Integration

None

### ODC Version

8.1.2

### Description

https://nvd.nist.gov/vuln/detail/CVE-2020-0187 was reported but this is against ` cpe:2.3:o:google:android:10.0:*:*:*:*:*:*:*` not pkg:maven/org.bouncycastle/bcprov-jdk15on@1.64

**Comment by aikebah** [comment]
This is a finding of the Sonatype OSSIndex, which indicates that their research shows (not surprisingly) that vanilla bouncycastle has the same flaw.
The issue is in (fixed in) the repackaged bouncycastle package included in android

**Comment by aikebah** [comment]
I suspect that the fix for it is part of bc 1.69 as that version added support for GCM-SIV and therefor is likely to no longer have the `startsWith("GCM")` that was at the root of the CVE. Validating that it is indeed the case is left up to you.

**Comment by jubui** [comment]
@aikebah you're saying that this is not a false positive? and that OSSIndex is right in that vanilla bouncycastle (meaning the bcprov-jdk15on library) also has the same problem, so you will not fix this in DependencyCheck. Is that a accurate summary of what you're saying?

**Comment by aikebah** [comment]
@jubui Without having looked too deep into it that is my impression indeed (based on a look at the android fix commit, the OSSIndex reporting it as affected CVE for bouncycastle and the fact that the changelog of bouncycastle shows introduction of GCM-CIV support in 1.69)

**Comment by aikebah** [comment]
In any case we will not suppress it. If you still believe it to be a false positive than it would be something to take up with the folks at OSSIndex. But at first glance I would expect this to be a true positive.

**Comment by jubui** [comment]
@aikebah Though based on my findings it looks like this is a false positive that only affects Android, I'm not 100% on that. I'm including my research with limited understanding.

```
I suspect that the fix for it is part of bc 1.69 as that version added support for GCM-SIV and therefor is likely to no longer have the startsWith("GCM") that was at the root of the CVE. Validating that it is indeed the case is left up to you.
```

Yes, I found that in bc 1.67 (not 1.69) a similar fix was made https://github.com/bcgit/bc-java/issues/767 which specifically calls out CVE-2020-0187.

This comment https://github.com/bcgit/bc-java/issues/767#issuecomment-683207432 sheds some light on why the issue was reported on android but not on bc. I am not 100% sure I'm interpreting this correctly, but it sounds like this is more of an android issue, not bc, but even the author seems to admit that they don't fully understand the issue.

**Comment by aikebah** [comment]
Sounds reasonable to share that with the OSSIndex team to have them re-evaluate their listing of bouncycastle as affected by the CVE at first read.

DependencyCheck is designed to simply report on what the various sources (of which OSSIndex is one, assuming that you leave the OSSIndexAnalyzer enabled) have registered as known vulnerabilities. So when OSSIndex stops returning the CVE for bouncycastle DependencyCheck would stop reporting it.

Contact information for OSSIndex can be found at https://ossindex.sonatype.org/doc/report-vulnerability
